lua/entities/starfall_prop/init.lua

```diff
@@ -15,6 +15,18 @@ function ENT:Initialize()
 	self:AddEFlags( EFL_FORCE_CHECK_TRANSMIT )
 end
 
+function ENT:PhysicsSimulate(physObj, dt)
+	if self.customForceMode then
+		return self.customForceAngular, self.customForceLinear, self.customForceMode
+	elseif self.customShadowForce then
+		self.customShadowForce.deltatime = dt
```

Comment:
Does `self.whatever` trigger __index calls? We might want to optimize against that for this function.

Reply:
Well it would only trigger it maximum of two times here on `self.customForceMode` and `self.customShadowForce`. I guess I could set `self.customPhysicsMode` to a number and limit the maximum possible `__index` lookup to one.  
  
Setting `self.customPhysicsMode` to `0` on entity initialization *(viable, because it only applies to `starfall_processor` and `starfall_prop`)* could possibly elimate the `__index` lookup altogether, however now that `StopMotionController` is used, this function simply stops being executed, the `else` statement below is just for safety in case it doesn't early enough.

Reply:
Sorry kinda sleepy, updated my comment now.